File: BOJ/11000-12000/11723/11723.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int n;

    string command;
    int num;
    int s = 0;

    cin >> n;

    for (int i = 0; i < n; ++i)
    {
        command.clear();
        cin >> command;

        if (command == "all" || command == "empty")
        {
            if (command == "all")
                s = (1 << 21) - 1;
            else
                s = 0;
        }
        else
        {
            cin >> num;

            char ch = command[0];

            if (ch == 'a')
                s |= 1 << num;
            else if (ch == 'c')
                if (s & (1 << num))
                    cout << 1 << "\n";
                else
                    cout << 0 << "\n";
            else if (ch == 'r')
                s &= ~(1 << num);
            else if (ch == 't')
                s ^= 1 << num;
        }
    }
}

File: BOJ/11000-12000/11723/11723.py
import sys

N = int(input())

S = [0 for i in range(20)]


def command_all(S):
    for i in range(20):
        S[i] = 1


def command_empty(S):
    for i in range(20):
        S[i] = 0


def command_toggle(S, x):
    if S[x-1] > 0:
        S[x-1] = 0
    else:
        S[x-1] = 1


def command_add(S, x):
    S[x-1] = 1


def command_remove(S, x):
    if S[x-1] > 0:
        S[x-1] = 0


def command_check(S, x):
    if S[x-1] == 1:
        print(1)
    else:
        print(0)


op_func = {
    "add": command_add,
    "remove": command_remove,
    "check": command_check,
    "toggle": command_toggle,
    "all": command_all,
    "empty": command_empty
}

for _ in range(N):
    command = sys.stdin.readline().split()

    if len(command) == 1:
        op_func[command[0]](S)
    else:
        op_func[command[0]](S, int(command[1]))
